Stop archiving gfsarch.log as it is being written

WalterKolczynski-NOAA opened the pull request:
The gfs archive job was failing because it was attempting to archive its own log file into gfsa.tar while it was being written. To exclude that file pattern, bash extended globbing is turned on, which allows the use of a negating group.

Fixes: #558

WalterKolczynski-NOAA committed:
Stop archiving gfsarch.log as it is being written

The gfs archive job was failing because it was attempting to archive
its own log file into gfsa.tar while it was being written. To exclude
that file pattern, bash extended globbing is turned on, which allows
the use of a negating group.

Fixes: #558

WalterKolczynski-NOAA commented:
@DavidHuber-NOAA Can you test this on one of your failures so I don't have to run several cycles just to test this?

DavidHuber-NOAA reviewed (changes requested):
This isn't working for me on S4.  Since these are `ksh` scripts, shopt is not a valid command.  However, extended globbing should already be part of the available syntax in `ksh`.  I've verified this on the terminal, so I'm not sure why it fails in the script.  I've uploaded the log file to Hera: /scratch1/NESDIS/nesdis-rdo2/David.Huber/s4_logs/384_p8a/gfsarch.log.

aerorahul commented:
@WalterKolczynski-NOAA 
We want to know exactly what is being archived.
Globbing can lead to archiving stale or irrelevant files or remnants.

aerorahul reviewed (approved):
Since this is a bugfix, I am fine with this change.
@WalterKolczynski-NOAA 
Mind if you change the interpreter in `ush/hpssarch_gen.sh` to `#/bin/bash` instead of `ksh`?

WalterKolczynski-NOAA committed:
Convert archive script to bash

The archive script is updated to bash to enable extended globbing. Due
to the shell change, base-10 needed to be specified for numbers that
may have leading zeros.

Refs: #558

WalterKolczynski-NOAA commented:
@DavidHuber-NOAA I changed the shell for the archive job to bash and it solves the problem for me on Hera. Please confirm it works for you.

DavidHuber-NOAA commented:
> @DavidHuber-NOAA I changed the shell for the archive job to bash and it solves the problem for me on Hera. Please confirm it works for you.

@WalterKolczynski-NOAA Yes, it works for me now.  Thanks!

DavidHuber-NOAA reviewed (approved):
This works for me now.
